File: Competetive programming/Python/Unit 3/WERTYU.py
keyboard = "`1234567890-=WERTYUIOP[]\\SDFGHJKL;'XCVBNM,./"

print("===== WERTYU =====\n")

print("Enter text:")
text = input()

result = ""

for ch in text:

    if ch == ' ':
        result += ' '

    else:
        pos = keyboard.find(ch)

        if pos > 0:
            result += keyboard[pos - 1]
        else:
            result += ch

print("\nCorrected Text:")
print(result)

File: Competetive programming/C/Unit 3/WERTYU.c
#include <stdio.h>
#include <string.h>
int main() {
    char keyboard[] =
    "`1234567890-=WERTYUIOP[]\\SDFGHJKL;'XCVBNM,./";

    char text[200];

    printf("===== WERTYU =====\n\n");

    printf("Enter text:\n");
    printf("Example: O S, GOMR YPFSU?\n\n");

    fgets(text, sizeof(text), stdin);

    printf("\nDecoded Text:\n");

    for(int i = 0; text[i] != '\0'; i++) {

        if(text[i] == ' ') {
            printf(" ");
            continue;
        }

        for(int j = 0; keyboard[j] != '\0'; j++) {

            if(text[i] == keyboard[j]) {
                printf("%c", keyboard[j - 1]);
                break;
            }
        }
    }

    return 0;
}